client/src/main/java/io/quarkus/vault/client/auth/VaultToken.java

```diff
@@ -5,42 +5,52 @@
 import java.time.Duration;
 import java.time.Instant;
 import java.time.InstantSource;
+import java.util.concurrent.atomic.AtomicInteger;
 
+import io.quarkus.vault.client.VaultException;
 import io.quarkus.vault.client.logging.LogConfidentialityLevel;
 
 public class VaultToken extends VaultTimeLimited {
 
     private final String clientToken;
     private final boolean fromCache;
+    private final AtomicInteger allowedUsesRemaining;
 
-    public static VaultToken from(String clientToken, boolean renewable, Duration leaseDuration, InstantSource instantSource) {
-        return new VaultToken(clientToken, renewable, leaseDuration, false, instantSource);
+    public static VaultToken from(String clientToken, boolean renewable, Duration leaseDuration, Integer allowedUsesRemaining,
+            InstantSource instantSource) {
+        return new VaultToken(clientToken, renewable, leaseDuration, allowedUsesRemaining, false, instantSource);
     }
 
-    public static VaultToken renewable(String clientToken, Duration leaseDuration, InstantSource instantSource) {
-        return from(clientToken, true, leaseDuration, instantSource);
+    public static VaultToken renewable(String clientToken, Duration leaseDuration, Integer allowedUsesRemaining,
+            InstantSource instantSource) {
+        return from(clientToken, true, leaseDuration, allowedUsesRemaining, instantSource);
     }
 
     public static VaultToken neverExpires(String clientToken, InstantSource instantSource) {
-        return from(clientToken, false, Duration.ofSeconds(Long.MAX_VALUE), instantSource);
+        return from(clientToken, false, Duration.ofSeconds(Long.MAX_VALUE), null, instantSource);
     }
 
-    public VaultToken(String clientToken, boolean renewable, Duration leaseDuration, boolean fromCache,
+    public VaultToken(String clientToken, boolean renewable, Duration leaseDuration, Integer allowedUsesRemaining,
+            boolean fromCache,
             InstantSource instantSource) {
         super(renewable, leaseDuration, instantSource);
         this.clientToken = clientToken;
         this.fromCache = fromCache;
+        this.allowedUsesRemaining = new AtomicInteger(
+                allowedUsesRemaining != null && allowedUsesRemaining > 0 ? allowedUsesRemaining : Integer.MAX_VALUE);
```

Comment:
why this condition `allowedUsesRemaining > 0` ?
if `allowedUsesRemaining <= 0`, then the remaining allowed uses should really be `MAX_VALUE`?

Reply:
The value is either `0` (which is unlimited) or the number of allowed uses, I believe the null check was just me being pedantic.

In either case we have to map `0`/`null` to something or we'd have to check `allowedUsesRemaining != null` before every access. This seemed like the cleaner method.

Reply:
I can see it now in https://developer.hashicorp.com/vault/api-docs/auth/token#num_uses
seemed strange that 0 uses means unlimited.